Hw06 email

Alex-T-T opened the pull request.

Alex-T-T committed:
send email message

1) send email on registration
2) resend email on filed first email

Alex-T-T committed:
replace sgMail

replace sgMail to folder Utils => sendEmail.js

Alex-T-T committed:
update sendEmail

Alex-T-T committed:
Create Dockerfile

OnyshchenkoO commented on src/controllers/authController.js:
```diff
@@ -23,7 +24,18 @@ const registerController = async (req, res, next) => {
+
+    const newUser = await User.create({ email, password: hashPassword, subscription, avatarURL, verificationToken });
+
+    const message = {
+        to: email,
+        subject: 'add email',
+        text: `All will be fine! Click here to verify your email: <a href='http://localhost:3000/api/users/verify/:${verificationToken}' target='_blank' >Click on me</a>`,
+        html: `<h1>All will be fine! Click here to verify your email: <a href='http://localhost:3000/api/users/verify/:${verificationToken}' target='_blank' >Click on me</a></h1>`
+    };
+
+    sendEmail(message);
```
У тебе цей код дублюється нижче. Винеси його в окремий файл

Alex-T-T replied:
виніс до файлу verificationMessage.js в папці src/utils

OnyshchenkoO commented on src/utils/sendEmail.js:
```diff
@@ -0,0 +1,17 @@
+sgMail.setApiKey(SENDGRID_API_KEY);
+
+const sendEmail = async (data) => {
+    const email = {...data, from: '0504Fusion@gmail.com'}
```
Винеси пошту в .env

Alex-T-T replied:
створив перемінну SENDER_EMAIL

Alex-T-T committed:
after mentor's review

1) create 'verificationMessage.js' => 'src/utils' and replace const message from 'authController.js'
2) add to '.env' SENDER_EMAIL

Alex-T-T committed:
Update verificationMessage.js
